### PR TITLE
Add SAMPLING_STORAGE_TYPE environment variable

### PR DESCRIPTION
**Pull request by joe-elliott**
## Which problem is this PR solving?
- Adds support for a new environment variable `SAMPLING_STORAGE_TYPE`. This allows the user to independently configure
their `SPAN_STORAGE_TYPE` and `SAMPLING_STORAGE_TYPE`. e.g. the user could choose to store their spans in elasticsearch while putting their adaptive sampling data in cassandra. This also paves the way for simpler to operate sampling storage types like redis or memcached.

## Short description of the changes
- Add support for `SAMPLING_STORAGE_TYPE` to plugin/storage/factory_config.go
- Add support to plugin/storage/factory.go to return the correct factory or fall back to previous behavior if unspecified.
- Add relevant testing.

### CODE REVIEW

**Comment by yurishkuro** [comment]
Just an observation - this will be subject to the existing limitation that a given kind of storage (e.g. cassandra) can only be configured once, i.e. it's not possible to have span and sampling storage use different instances of cassandra (or even different namespaces).

I don't think it should block this PR, but it may be worth thinking how that can be improved. For example, we have the mechanism for configuring different "named" storage, namely "archive", via flags like `--es.archive.***`. We could extend that mechanism to storage types, e.g. `--es.dependencies.***` or `--es.sampling.***`.

Of course, if the new solution prevents sharing the same instance as we can today that would be a regression.

**Review by yurishkuro — APPROVED** [review]

File: plugin/storage/factory_config_test.go [hunk 1]
```diff
@@ -37,15 +38,18 @@ func TestFactoryConfigFromEnv(t *testing.T) {
 	assert.Equal(t, cassandraStorageType, f.SpanWriterTypes[0])
 	assert.Equal(t, cassandraStorageType, f.SpanReaderType)
 	assert.Equal(t, cassandraStorageType, f.DependenciesStorageType)
+	assert.Equal(t, "", f.SamplingStorageType)
```
Comment (albertteoh) [inline plugin/storage/factory_config_test.go:41]: nit: could use `assert.Empty(t, f.SamplingStorageType)`

**Review by albertteoh — APPROVED** [review]
LGTM. Does this mean we need to add support for `SAMPLING_STORAGE_TYPE` in jaeger-operator similar to https://github.com/jaegertracing/jaeger-operator/issues/1700?

**Comment by joe-elliott** [comment]
> LGTM. Does this mean we need to add support for SAMPLING_STORAGE_TYPE in jaeger-operator similar to https://github.com/jaegertracing/jaeger-operator/issues/1700?

Oof, I don't know much about the operator. I had intended to PR the docs, but can't say I feel confident in my ability to make a well thought out operator change.

> this will be subject to the existing limitation that a given kind of storage (e.g. cassandra) can only be configured once, i.e. it's not possible to have span and sampling storage use different instances of cassandra (or even different namespaces).

This is a good point. I will include a note about this in my docs PR.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**